src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -0,0 +1,24 @@
+package pureconfig
+
+trait ConfigFieldMapping[T] {
```

Comment:
For comparison, what would this (or the former approach) look like to acheivehyphen-delimiter style throughout the config?